platform/ios/src/MGLMapView.mm

```diff
@@ -750,6 +753,20 @@ - (void)glkView:(__unused GLKView *)view drawInRect:(__unused CGRect)rect
         _mbglMap->render();
 
         [self updateUserLocationAnnotationView];
+        
+        // Hide all annotation views
+        for (MGLAnnotationView *annotationView in _annotationViews) {
+            annotationView.hidden = YES;
+        }
+        
+        // Update visible annotation views
+        std::vector<MGLAnnotationTag> annotationTags = [self annotationTagsInRect:CGRectInset(self.bounds, 0, 0)];
+        for (std::size_t i = 0; i != annotationTags.size(); ++i) {
```

Comment:
Thanks. I think that, in this case, I need the `i` since I want to take the intersection of the `annotationTagsInRect` and the annotation tags in `_annotationContextsByAnnotationTag`. I want to get the context from `_annotationContextsByAnnotationTag` with the key of `annotationTags[i]`. Only those visible contexts need to have their view update to reflect where they should now show on the map. I will use the `for (auto &pair ...` expression in my next commit for view reuse though. Please stay tuned!